File: Exercises/ExercisesFortran/11Jan_Chapter07/4-BarrierCrossing/Source/mdloop.f
      Subroutine Mdloop
      Implicit None
 
      Include 'system.inc'

      Integer I,J,CycleMultiplication
      Double Precision Ran_Vel,EnergyDrift,CumEnergyDrift,CumCycles

      Parameter (CycleMultiplication = 1000)

      Call Sample(1,0)

      CumEnergyDrift = 0.0d0
      CumCycles = 0.0d0

      Do I=1,Ncycle
         Do J=1,CycleMultiplication

Ccccccccccccccccccccccccccccccccccccccccccccccccc
C     Generate Initial Coordinates              C
C                                               C
C     Perform CycleMultiplication*Ncycle Md     C
C     Simulations With Different Initial        C
C     Conditions.                               C
C                                               C
C     Xpos   = Starting Position                C
C     Vpos   = Starting Velocity                C
C     Theta  = Storage For Initial Velocity     C
C     Qstar  = Place Of The Dividing Surface    C
C                                               C
C     To Program:                               C
C                                               C
C     -Generate Initial Position/Velocity       C
C     -Integrate The Equations Of Motion By A   C
C      Function Call To Subroutine Integrate    C
C     -Beware That In Integrate The Subroutine  C
C      Sample Is Called !!                      C
C                                               C
C     The Averaged Energy Drift (Over All       C
C     Simulations) Equals CumEnergyDrift/CumCycles                 C
Ccccccccccccccccccccccccccccccccccccccccccccccccc

C Start Modification


C End Modification

         Enddo
      Enddo

      Call Sample(3,0)

      Write(6,*) 'Av. Energy Drift      :',CumEnergyDrift/CumCycles

      Return
      End

File: Exercises/ExercisesC/11Jan_Chapter07/4-BarrierCrossing/Source/mdloop.c
#include <stdio.h>
#include <stdlib.h>
#include <stdlib.h>
#include <math.h>
#include "system.h"
#include "ran_uniform.h"

#define CycleMultiplication 1000

void MdLoop(void)
{
  int i,j;
  double EnergyDrift,CumEnergyDrift,CumCycles;

  Sample(1,0);

  CumEnergyDrift=0.0;
  CumCycles=0.0;

  for(i=0;i<NumberOfCycles;i++)
    for(j=0;j<CycleMultiplication;j++)
    {
      // generate initial coordinates
      //
      // perform 1000*Ncycle MD simulations with 
      // different initial conditions
      //
      // Xpos   = starting position
      // Vpos   = starting velocity
      // Theta  = storage for initial velocity
      // Qstar  = place of the dividing surface
      //
      // To Program:
      //
      // -Generate Initial Position/Velocity
      // -Integrate The Equations Of Motion By A 
      //  Function Call To Subroutine Integrate 
      // -Beware That In Integrate The Subroutine
      //  Sample Is Called !!
      //
      // The Averaged Energy Drift (Over All
      // Simulations) Equals CumEnergyDrift/CumCycles

      // start modification

      // end modification

    }

  Sample(3,0);
  printf("Av. Energy Drift      : %f\n",CumEnergyDrift/CumCycles);
}
